tests/root_io/read_interface.cpp

```diff
@@ -0,0 +1,110 @@
+    if (frame.get("emptySubsetColl") == nullptr) {
+      std::cerr << "Could not retrieve an empty subset collection" << std::endl;
+      return 1;
+    }
+    if (frame.get("emptyCollection") == nullptr) {
+      std::cerr << "Could not retrieve an empty collection" << std::endl;
+      return 1;
+    }
+
+    processEvent(frame, i, reader.currentFileVersion());
+
+    auto otherFrame = reader.readNextFrame("other_events");
+    processEvent(otherFrame, i + 100, reader.currentFileVersion());
+    // The other_events category also holds external collections
+    processExtensions(otherFrame, i + 100, reader.currentFileVersion());
+    // As well as a test for the vector members subset category
+    checkVecMemSubsetColl(otherFrame);
+  }
+
+  // if (reader.readNextFrame(podio::Category::Event)) {
+  //   std::cerr << "Trying to read more frame data than is present should return a nullptr" << std::endl;
+  //   return 1;
+  // }
+
+  std::cout << "========================================================\n" << std::endl;
+  // if (reader.readNextFrame("not_present")) {
+  //   std::cerr << "Trying to read non-existant frame data should return a nullptr" << std::endl;
+  //   return 1;
+  // }
+
+  // Reading specific (jumping to) entry
+  {
+    auto frame = reader.readFrame(podio::Category::Event, 4);
+    processEvent(frame, 4, reader.currentFileVersion());
+    // Reading the next entry after jump, continues from after the jump
+    auto nextFrame = reader.readNextFrame(podio::Category::Event);
+    processEvent(nextFrame, 5, reader.currentFileVersion());
+
+    auto otherFrame = reader.readFrame("other_events", 4);
+    processEvent(otherFrame, 4 + 100, reader.currentFileVersion());
+    if (reader.currentFileVersion() > podio::version::Version{0, 16, 2}) {
+      processExtensions(otherFrame, 4 + 100, reader.currentFileVersion());
+    }
+
+    // Jumping back also works
+    auto previousFrame = reader.readFrame("other_events", 2);
+    processEvent(previousFrame, 2 + 100, reader.currentFileVersion());
+    if (reader.currentFileVersion() > podio::version::Version{0, 16, 2}) {
+      processExtensions(previousFrame, 2 + 100, reader.currentFileVersion());
+    }
+
+    // Trying to read a Frame that is not present returns a nullptr
+    // if (reader.readFrame(podio::Category::Event, 10)) {
+    //   std::cerr << "Trying to read a specific entry that does not exist should return a nullptr" << std::endl;
+    //   return 1;
+    // }
+  }
+
+  return 0;
+}
+
+int main(int, char**) {
+
+  auto reader = podio::makeReader("example_frame_interface.root");
+  if (read_frames(reader)) {
+    return 1;
+  }
+
+#if PODIO_ENABLE_RNTUPLE
+  auto readerRNTuple = podio::makeReader("example_frame_rntuple_interface.root");
+  if (read_frames(readerRNTuple)) {
+    return 1;
+  }
+#endif
+
+#if PODIO_ENABLE_SIO
+  auto readerSIO = podio::makeReader("example_frame_sio_interface.sio");
+  if (read_frames(readerSIO)) {
+    return 1;
+  }
+#endif
```

Comment:
This should go into the `sio_io` tests directory, also for the writing. That might require shuffling some of the code in the .cpp files into headers and putting them into the top level `tests` directory.

Reply:
I added some more general includes in the top tests directory and then the actual tests in `tests/root_io` and `tests/sio_io`. I've added a new library `podioIO` since I don't think there is a good place for the reader and writer (but then one will need to link to this one to use the interfaces). I think this should eventually replace the current instances since the interface should cover most of the use cases and you can also change backend very easily.

Reply:
About the new library / target. I think this should be the new default target to link against. It brings along all the correct dependencies in any case (it links publicly against `podioRootIO` and potentially also `podioSioIO`). So users should just use `podioIO` in their setups. We will have to document that somewhere. However, I don't think we have a fitting section yet to put it into. I only ever put the cmake macros on to slides so far, it seems.